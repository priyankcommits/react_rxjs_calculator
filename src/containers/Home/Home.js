// Home Container Component
import Col from 'react-bootstrap/Col';
import Container from 'react-bootstrap/Container'; 
import React from 'react';
import Row from 'react-bootstrap/Row';

import  { calculatorButtons } from '../../shared/constants/index';
import { BoxStyle, HomeStyle } from '../../shared/styles/main/index';
import { ButtonLayout, History, Information, InputBox, OutputBox } from '../../components/index';

const Home = () => {

  return (
    <Container as={HomeStyle}>
      <Row className='mt-1'>
        <h3>Calculator using React Hooks + RxJS</h3>
        <Information buttons={calculatorButtons} />
      </Row>
      <Row className='mt-3'>
        <Col md={8}>
          <BoxStyle className='p-1'>
            <Row>
              <Col xs={1}>
                <History />
              </Col>
              <Col md={6}>
                <InputBox />
              </Col>
              <Col md={5}>
                <OutputBox />
              </Col>
              <Col md={12}>
                <ButtonLayout buttons={calculatorButtons} />
              </Col>
            </Row>
          </BoxStyle>
        </Col>
      </Row>
    </Container>
  );
}

export { Home };
