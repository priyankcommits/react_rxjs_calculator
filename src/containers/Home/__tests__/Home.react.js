// Snapshot testing
import React from 'react';
import renderer from 'react-test-renderer';

import { Home } from '../Home';

test('Home Initial Rendering', () => {
  const component = renderer.create(
    <Home />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
