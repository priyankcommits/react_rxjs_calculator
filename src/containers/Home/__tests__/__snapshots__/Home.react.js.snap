// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Initial Rendering 1`] = `
<div
  className="sc-AxjAm jNmcQG container"
>
  <div
    className="mt-1 row"
  >
    <h3>
      Calculator using React Hooks + RxJS
    </h3>
    <div
      className="sc-AxhCb kgMfC mt-1 ml-2 btn btn-dark btn-sm"
      disabled={false}
      onClick={[Function]}
    >
      <span
        className="mx-1"
      >
        i
      </span>
    </div>
  </div>
  <div
    className="mt-3 row"
  >
    <div
      className="col-md-8"
    >
      <div
        className="sc-AxirZ bQmHBL p-1"
      >
        <div
          className="row"
        >
          <div
            className="col-1"
          >
            <div
              className="sc-AxhCb kgMfC mt-1 btn btn-dark btn-sm"
              disabled={false}
              onClick={[Function]}
            >
              <span>
                H
              </span>
            </div>
          </div>
          <div
            className="col-md-6"
          >
            <div
              className="sc-AxirZ bQmHBL"
            >
              <div
                className="sc-AxiKw eGDepq"
              >
                <h4>
                  
                  <span
                    className="text-danger"
                  >
                    
                  </span>
                </h4>
              </div>
            </div>
          </div>
          <div
            className="col-md-5"
          >
            <div
              className="sc-AxirZ bQmHBL"
            >
              <div
                className="sc-AxiKw eGDepq"
              >
                <h4>
                  
                </h4>
              </div>
            </div>
          </div>
          <div
            className="col-md-12"
          >
            <div>
              <div
                className="mt-2 row"
              >
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-secondary"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    sin
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-secondary"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    ln
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-light"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    7
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-light"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    8
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-light"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    9
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-secondary"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    รท
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-secondary"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    cos
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-secondary"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    log
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-light"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    4
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-light"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    5
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-light"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    6
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-secondary"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    x
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-secondary"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    tan
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-secondary"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    ฯ
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-light"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    1
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-light"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    2
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-light"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    3
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-secondary"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    -
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-secondary"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    pow
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-secondary"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    e
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-light"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    0
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-light"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    .
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-primary"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    AC
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-secondary"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    +
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-secondary"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    (
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-secondary"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    )
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-warning"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    Undo
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-warning"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    Redo
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-primary"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    Ans
                  </button>
                </div>
                <div
                  className="my-1 col-2"
                >
                  <button
                    className="w-100 btn btn-secondary"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    %
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
